lambdas/functions/data_transfer_manager/data_transfer_manager.py

```diff
@@ -254,13 +255,31 @@ def handler(event, context):
             f",Error: {e}",
         }
 
+    logger.info("Submit AWS Batch Job list:")
+    logger.info(json.dumps(batch_job_data))
+
+    logger.info("Update Dynamodb list:")
+    logger.info(
+        json.dumps([item.__dict__ for item in dynamodb_job], cls=util.JsonSerialEncoder)
+    )
+
     # Submit Batch jobs
     if event.get("skip_submit_batch_job"):
         logger.info("Skip submit batch job flag is raised. Skipping ...")
     else:
-        logger.info(f"Batch Job list: {json.dumps(batch_job_data, indent=4)}")
-        for job_data in batch_job_data:
+        logger.info("Submitting batch job")
+        for i, job_data in enumerate(batch_job_data):
             submit_data_transfer_job(job_data)
+
+            # Sleep for 1 second every 10th job
+            # This is to prevent AWS Batch SubmitJob throttling limit of 50 jobs per second
+            # Putting 10 here as it is shared across 3 reserved concurrency limit, and
+            # 2 lambda function (transfer-manager and validation-manager)
+            # Without sleep, the average of submitting jobs is about 13 jobs per second
+            # https://docs.aws.amazon.com/batch/latest/userguide/service_limits.html
+            if (i + 1) % 15 == 0:
```

Comment:
15? Or 10? 
Above says 10...
Also, with 2 lambdas, each having a concurrency limit of 3, that's 6 lambdas submitting jobs in parallel, no? Would that not mean even with a limit of 10 jobs/sec they could reach the limit of 50/sec?

Reply:
Yes, did update it on [04a78c0](https://github.com/umccr/agha-data-validation-pipeline/pull/59/commits/04a78c0c1519d3a8e346a9244261615e89a4f12a)

Reply:
All good! 
Also noticed your comment that the limit could still be reached, but it's unlikely.

Reply:
Happy to give it a shot.

Reply:
I calculated on average that it took about 13 jobs per second, so I suppose having it as 10 should be enough to give a wait. And I just thought since it is less likely that both lambdas run at the same time, so it is unlikely to happen.

But I guess we could bring it down to (50/6=) 8 jobs per second per lambda